Server/mods/deathmatch/logic/CGame.cpp

```diff
@@ -4699,3 +4705,31 @@ void CGame::HandleCrashDumpEncryption()
     }
 #endif
 }
+
+void CGame::RegisterClientTriggeredEventUsage(CPlayer* pPlayer)
+{
+    if (!pPlayer)
+        return;
+
+    auto iter = m_mapClientTriggeredEvents.find(pPlayer);
+
+    if (iter == m_mapClientTriggeredEvents.end())
+        iter = m_mapClientTriggeredEvents.insert({pPlayer, 0}).first;
+
+    iter->second++;
+}
+
+void CGame::ProcessClientTriggeredEventSpam()
+{
+    for (const auto& pair : m_mapClientTriggeredEvents)
```

Comment:
Additionally, server side deatchmatch project has C++17 enabled. And you can use the structured binding: `for (const auto& [player, count] : m_mapClientTriggeredEvents)`